middleware/assert_test.go

```diff
@@ -14,106 +14,149 @@ import (
+		}, nil
+	}
+	datastoreMock.GetFilterFunc = func(ctx context.Context, filterID, etag string) (*models.Filter, error) {
+		filter := &models.Filter{
+			Dataset: &models.Dataset{
+				Version: 1,
+			},
+			ID: filterID,
+		}
+		if filterIsFlexible {
+			filter.Type = "flexible"
 		}
+		return filter, nil
+	}
+	return datastoreMock
+}
 
-		filterFlexAPIMock := &apimock.FilterFlexAPIMock{
-			ForwardRequestFunc: func(r *http.Request) (*http.Response, error) {
-				return expectedResponse, nil
-			},
+func setupDatasetAPIMock(datasetIsFlexible bool) *apimock.DatasetAPIMock {
+	datasetAPIMock := &apimock.DatasetAPIMock{}
+
+	datasetAPIMock.GetFunc = func(ctx context.Context, ut, st, cid, dsid string) (dataset.DatasetDetails, error) {
+		dataset := dataset.DatasetDetails{
+			Type: "not_flexible",
 		}
+		if datasetIsFlexible {
+			dataset.Type = "cantabular_flexible_table"
+		}
+		return dataset, nil
+	}
+	return datasetAPIMock
+}
 
-		Convey("When a filter with given ID in the datastore of type 'flexible' is found", func(){
-			datastoreMock := mock.NewDataStore().Mock
-			datastoreMock.GetFilterFunc = func(ctx context.Context, filterID, etag string) (*models.Filter, error) {
-				return &models.Filter{
-					Type: "flexible",
-					Dataset: &models.Dataset{
-						Version: 1,
-					},
-					ID: filterID,
-				}, nil
-			}
-
-			Convey("When an incoming request passes through the assert.FilterType middleware", func(){
-				assert := middleware.NewAssert(
-					responder.New(),
-					&mock.DatasetAPI{},
-					filterFlexAPIMock,
-					datastoreMock,
-					testToken,
-					true,
-				)
+func assertTestSetup(filterIsFlexible, datasetIsFlexible, assertIsEnabled bool, testBluePrintID string) (*apimock.DataStoreMock, *apimock.FilterFlexAPIMock, *apimock.DatasetAPIMock, *middleware.Assert) {
+	filterFlexAPIMock := setupFilterFlexAPIMock()
+	datastoreMock := setupDataStoreMock(filterIsFlexible, testBluePrintID)
+	datasetAPIMock := setupDatasetAPIMock(datasetIsFlexible)
+
+	assert := middleware.NewAssert(
+		responder.New(),
+		datasetAPIMock,
+		filterFlexAPIMock,
+		datastoreMock,
+		testToken,
+		assertIsEnabled,
+	)
+	return datastoreMock, filterFlexAPIMock, datasetAPIMock, assert
+}
+
+func TestAssertFilterType(t *testing.T) {
+	Convey("Given a healthy dp-cantabular-filter-flex-api", t, func() {
+		testBluePrintID := ""
+		datasetIsFlexible := false
+		Convey("When a filter with given ID in the datastore of type 'flexible' is found", func() {
+			filterIsFlexible := true
+			Convey("When an incoming request passes through the assert.FilterType middleware", func() {
+				assertIsEnabled := true
+				datastoreMock, filterFlexAPIMock, _, assert := assertTestSetup(filterIsFlexible, datasetIsFlexible, assertIsEnabled, testBluePrintID)
 
 				w := httptest.NewRecorder()
 
 				testID := "12345678"
-				r, err := http.NewRequest("GET", "http://localhost:1234/test/" + testID, nil)
+				r, err := http.NewRequest("GET", "http://localhost:1234/test/"+testID, nil)
```

Comment:
Might be worth just getting rid of this, since it has no effect on the test. The vars are just injected by `SetURLVars`:

```go
r, err := http.NewRequest(http.MethodGet, "/", nil)
So(err, ShouldBeNil)
```

Reply:
Also any time you use `NewRequest` here you can use the `httptest` version, which panics if there is an error:

```go
r := httptest.NewRequest(http.MethodGet, "/", nil)
```

Reply:
Addressed.